plugins/javaembed/javaembed.cpp

```diff
@@ -1145,16 +1166,28 @@ class JavaRowStream : public CInterfaceOf<IRowStream>
     virtual void stop()
     {
         resultAllocator.clear();
-        if (iterator)
+        JNIEnv *JNIenv = queryJNIEnv();
+        if (JNIenv)
```

Comment:
I added it because now it is being called from destructor, and I was a little nervous that destruction COULD happen on a thread where JNIenv had not been set up.